website/server/models/group.js

```diff
@@ -577,6 +578,18 @@ schema.statics.cleanGroupQuest = function cleanGroupQuest () {
   };
 };
 
+async function _updateUserWithRetries (userId, updates, numTry = 1) {
+  return User.update({_id: userId}, updates).exec((err, raw) => {
```

Comment:
Sorry, this is just due to my lack of experience with Javascript. I've adjusted it to the promise syntax now and it's way better. I'm not using the .catch() method because it seems it doesn't work well with mongoose's promise library and so instead I'm using the .then(function(success), function(error)) syntax. Let me know if that works for you.